From 36c51649dcc175f28fd500c01ba8cdd596f15aa4 Mon Sep 17 00:00:00 2001
From: Masahiro Yamada <masahiro.yamada@canonical.com>
Date: Wed, 3 Dec 2025 17:12:26 +0900
Subject: [PATCH] listmount04: require kernel version 6.11

On Linux kernel 6.8, the listmount04 test fails as follows:

  Running tests.......
  tst_buffers.c:57: TINFO: Test is using guarded buffers
  tst_test.c:2025: TINFO: LTP version: 20250930
  tst_test.c:2028: TINFO: Tested kernel: 6.8.0 #71 SMP PREEMPT_DYNAMIC Wed Dec  3 16:00:27 JST 2025 x86_64
  tst_kconfig.c:88: TINFO: Parsing kernel config '/proc/config.gz'
  tst_kconfig.c:676: TINFO: CONFIG_TRACE_IRQFLAGS kernel option detected which might slow the execution
  tst_test.c:1846: TINFO: Overall timeout per run is 0h 02m 00s
  listmount04.c:128: TPASS: request points to unaccessible memory : EFAULT (14)
  listmount04.c:128: TPASS: mnt_ids points to unaccessible memory : EFAULT (14)
  listmount04.c:128: TPASS: invalid flags : EINVAL (22)
  listmount04.c:128: TPASS: insufficient mnt_id_req.size : EINVAL (22)
  listmount04.c:128: TPASS: invalid mnt_id_req.spare : EINVAL (22)
  listmount04.c:128: TFAIL: invalid mnt_id_req.param invalid retval 20: SUCCESS (0)
  listmount04.c:128: TFAIL: invalid mnt_id_req.mnt_id expected EINVAL: ENOENT (2)
  listmount04.c:128: TPASS: non-existant mnt_id : ENOENT (2)

  Summary:
  passed   6
  failed   2
  broken   0
  skipped  0
  warnings 0
  INFO: ltp-pan reported some tests FAIL
  LTP Version: 20250930
  INFO: Test end time: Wed Dec  3 07:01:33 UTC 2025

This test passes only after kernel commit 4bed843b1000 ("fs: reject
invalid last mount id early"), i.e. kernel version 6.11.

Link: https://lore.kernel.org/ltp/20251203081226.1148236-1-masahiro.yamada@canonical.com/
Fixes: 141958ad3e ("Add listmount04 test")
Reviewed-by: Petr Vorel <pvorel@suse.cz>
Signed-off-by: Masahiro Yamada <masahiro.yamada@canonical.com>
---
 testcases/kernel/syscalls/listmount/listmount04.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/testcases/kernel/syscalls/listmount/listmount04.c b/testcases/kernel/syscalls/listmount/listmount04.c
index a52bad064c0e..a6921a24975d 100644
--- a/testcases/kernel/syscalls/listmount/listmount04.c
+++ b/testcases/kernel/syscalls/listmount/listmount04.c
@@ -133,7 +133,7 @@ static void run(unsigned int n)
 static struct tst_test test = {
 	.test = run,
 	.tcnt = ARRAY_SIZE(tcases),
-	.min_kver = "6.8",
+	.min_kver = "6.11",
 	.bufs = (struct tst_buffers []) {
 		{ &request, .size = MNT_ID_REQ_SIZE_VER0 },
 		{},
-- 
2.52.0

